### PR TITLE
Progress Bar

### PR DESCRIPTION
**Pull request by umarmanzoor**
Progress Bar

### CODE REVIEW

**Comment by kordjamshidi** [comment]
@Rahgooy merge if you are ok with it.

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala [hunk 1]
```diff
@@ -138,7 +145,11 @@ class Node[T <: AnyRef](val keyFunc: T => Any = (x: T) => x, val tag: ClassTag[T
 
   /** Operator for adding a sequence of T into my table. */
   def populate(ts: Iterable[T], train: Boolean = true, populateEdge: Boolean = true) = {
+    progressBar.progress()
+    progressBar.init()
+    progressBar.total = ts.size;
     ts.foreach(addInstance(_, train, populateEdge))
```
Comment (Rahgooy) [inline saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala:151]: IMO, add progress-bar update to the `foreach` here not to the `addInstance` method.

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala [hunk 2]
```diff
@@ -106,7 +110,7 @@ class Node[T <: AnyRef](val keyFunc: T => Any = (x: T) => x, val tag: ClassTag[T
 
     if (containsNT(nodeInstance)) {
       logger.trace(s"The instance $instance is duplicate and it will be ignored! " +
-        s"This might be because you add the same instance to both train and test set. ")
+        s"This might be because you= add the same instance to both train and test set. ")
```
Comment (Rahgooy) [inline saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala]: correct this.

**Comment by Rahgooy** [comment]
great feature!
thanks.

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala (hunk 1 above)
Comment (umarmanzoor) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala:151]: I don't want to change the original code, that's why I put it in addInstance., I think let's leave it as it is...

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala (hunk 2 above)
Comment (umarmanzoor) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala]: @Rahgooy correct what? can you please explain?

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala (hunk 1 above)
Comment (Rahgooy) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala:151]: ok

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala (hunk 2 above)
Comment (Rahgooy) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala]: it seems you added a `=` to line 113, unintentionally.

Comment (kordjamshidi) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala]: Yaa and it also says `you add` :-)

Comment (umarmanzoor) [reply saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala]: ah, yes..

**Pull request merged**